torch/_torch_docs.py

```diff
@@ -8329,38 +8334,34 @@ def merge_dicts(*dicts):
                tensors for `U` and `Vh`, whereas :func:`torch.linalg.svd` returns
                empty tensors.
 
-.. note:: The singular values are returned in descending order. If :attr:`input` is a batch of matrices,
-          then the singular values of each matrix in the batch are returned in descending order.
-
 .. note:: The `S` tensor can only be used to compute gradients if :attr:`compute_uv` is `True`.
 
 .. note:: When :attr:`some` is `False`, the gradients on `U[..., :, min(m, n):]`
           and `V[..., :, min(m, n):]` will be ignored in the backward pass, as those vectors
           can be arbitrary bases of the corresponding subspaces.
 
-.. note:: The implementation of :func:`torch.linalg.svd` on CPU uses LAPACK's routine `?gesdd`
-          (a divide-and-conquer algorithm) instead of `?gesvd` for speed. Analogously,
+.. note:: The implementation of :func:`torch.linalg.svd` on CPU uses LAPACK's routine `gesdd`
+          (a divide-and-conquer algorithm) instead of `gesvd` for speed. Analogously,
           on GPU, it uses cuSOLVER's routines `gesvdj` and `gesvdjBatched` on CUDA 10.1.243
           and later, and MAGMA's routine `gesdd` on earlier versions of CUDA.
 
 .. note:: The returned `U` will not be contiguous. The matrix (or batch of matrices) will
           be represented as a column-major matrix (i.e. Fortran-contiguous).
 
-.. warning:: The gradients with respect to `U` and `V` will only be finite when the input does not
-             have zero nor repeated singular values.
-
-.. warning:: If the distance between any two singular values is close to zero, the gradients with respect to
-             `U` and `V` will be numerically unstable, as they depends on
-             :math:`\frac{1}{\min_{i \neq j} \sigma_i^2 - \sigma_j^2}`. The same happens when the matrix
-             has small singular values, as these gradients also depend on `S⁻¹`.
-
-.. warning:: For complex-valued :attr:`input` the singular value decomposition is not unique,
-             as `U` and `V` may be multiplied by an arbitrary phase factor :math:`e^{i \phi}` on every column.
+.. warning:: The singular values of a matrix are not unique. Any pair of left and right singular values may
+             be multiplied by `-1` in the real case or by a factor of norm `1` :math:`e^{i \phi}` in the complex case.
              The same happens when :attr:`input` has repeated singular values, where one may multiply
```

Comment:
I'm not sure how this paragraph connects to the previous paragraph. The previous paragraph states a mathematical property of the SVD. But then this paragraph begins with "The same happens when..." But what is "the same"?

Reply:
It comes to say that when the matrix has repeated singular values, there is a second source of "non uniqueness" for the singular vectors.

What's really happening here geometricaly is that SVD (and  `eigh`) work with subspaces, but we choose orthogonal bases to represent them.

If you have a linear subspace, if it's 1D (a line), it can be represented via an orthogonal basis choosing the vector of norm 1 in the direction of the subspace (the direction of the line). Of course, you have two choices, "point to the right" and "point to the left", that is, choosing `v` or choosing `-1*v = -v`.
If it has more dimensions, say it has k dimensions, you can choose a basis, v_1, ... , v_k, but you can rearrange it as v_2, v_1, v_3, ... , v_k, and it will still span the same subspace. Even more, you can rotate the vectors inside the subspace and the new set of rotated vectors will also be orthogonal and span the same subspace. This is this non-uniqueness.

In the complex case this is even worse, as a 1D complex subspace has two real dimension, so you have the problem with being able to "rotate" the basis even in 1D (that's what the multiplication by e^{i \phi}  is really doing).

Reply:
I tried to address this in a more explicit way in https://github.com/pytorch/pytorch/pull/56265#discussion_r615756832

Reply:
Combining these thoughts into the warning above makes them much clearer and logically connects the docs